### ISSUE TITLE
connectorSizing vs. conditional instantiation

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/ModelicaSpecification#444]
**Reported by AHaumer on 23 Oct 2010 06:44 UTC**
The definition of annotation(connectorSizing) is not very well considered. Have a look (is this legal according to the specification?):

```
model Test
  parameter Boolean useSine=true
    annotation(Evaluate=true);
  Modelica.Blocks.Math.MultiSum multiSum(nu=2);
  Modelica.Blocks.Sources.Ramp ramp(duration=1, height=2);
  Modelica.Blocks.Sources.Sine sine(freqHz=1) if useSine;
equation
  connect(ramp.y, multiSum.u[1]);
  connect(sine.y, multiSum.u[2]);
end Unnamed;
```

To me as a simple user it seems logical: when a conditionally instantiated component is removed (useSine = false in the above example), the connections to that component are removed. BUT: the connectorSizing - parameter is not adapted?

I dislike this "fancy" annotation, it brings up a lot of confusion. We have to clarify the specification.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/444

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 7 Mar 2012 14:09 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 15 Aug 2012 08:22 UTC**
Left-over tickets from the now closed [milestone:ModelicaSpec3.3] milestone which basically means it should go into [milestone:ModelicaSpec3.4] right?

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 30 Mar 2013 14:28 UTC**
The connectorSizing annotation does not handle all problems when connecting to a vector of connectors. This is a compromise to get convenient functionality in most cases, but not in all. The current design is that of an annotation, that is a tool can completely ignore it. Furthermore, the design is that this annotation triggers an action at the time instant when connecting. Then, a modifier to the vectorized connector is updated and the index of the connection line is updated. Afterwards, the annotation has no longer an effect, and especially it does not have any effect during translation.

In the situation sketched in the ticket, the dimension of the vectorized connector has to be adapted based on the actual value of a parameter. This can only be performed at translation time. This means the desired functionality cannot be provided with an annotation, but must be provided with a language element.

To summarize: It is impossible to provide the desired functionality with an annotation, it must be a language element. Therefore, the assumption that the definition of the ConnectorSizing annotation shall be improved is wrong, and therefore also this ticket.

If you want the behavior that you describe a new language construct has to be introduced. The question is whether this is worth the effort since this will become tricky and is only needed in some corner cases.

In the ticket example, the modeler anyway has to model in the text layer. With current tools, he/she cannot do it purely in a graphical way. However, then it is fine that the modeler has to define the connection index and the connector dimension also by him/herself.

**Comment by modelica-trac-importer** [comment]
**Changelog removed by otter on 30 Mar 2013 14:28 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by ahaumer on 20 Dec 2016 16:52 UTC**
